### ISSUE TITLE
[Intern Progress] Abdullah Akintobi - Ersilia Outreachy Contributions

### ISSUE THREAD

**Issue by abdullahakintobi** [ersilia-os/outreachy-contributions#17]
# 🌟 Intern Progress Tracker  
Welcome! Use this issue to track your progress as you work through the AI/ML modeling exercise. Follow each step, check the boxes when completed, and update your issue as you go.

# ▶️ **Entry tasks**
- [x] Motivation letter posted in this issue
- [x] Installed the Ersilia Model Hub and successfully run models!
- [x] Recorded system settigns in this issue
- [x] Explained completed tests of Ersilia in the issue

---

##  **1. Download a Dataset of Interest**
- [x] Created a fork of this repository to work on
- [ ] Selected a dataset from Therapeutics Data Commons (TDC)
- [ ] Verified that the dataset is a classification problem*
- [ ] Read and understood the dataset’s background and endpoint  
- [ ] Assessed computational feasibility (size, features, etc.)  
- [ ] Downloaded the dataset using Python  
- [ ] Saved data in the `/data` folder  
- [ ] Started writing documentation in the README (Installation, dataset details, how to run code)  

---

##  **2. Featurise the Data**
- [ ] Browsed Ersilia Model Hub for "Representation" models  
- [ ] Selected an appropriate featuriser and justified my choice  
- [ ] Implemented featurisation in a notebook or script
- [ ] Ensured the process is reproducible 
- [ ] Updated the README with featurisation instructions  

---

##  **3. Build an ML Model**
- [ ] Selected an ML framework
- [ ] Trained the model on featurised data  
- [ ] Evaluated the model (accuracy, precision, recall, etc.)  
- [ ] Created graphs using Matplotlib for better visualization  
- [ ] Ensured all steps are reproducible  
- [ ] Documented the model-building process in README

---

##  **4. Prepare Code for Review**
- [ ] Double-checked that documentation is clear and complete  
- [ ] Evaluated if my model performance is satisfactory 
- [ ] Identified possible improvements  
- [ ] Updated this issue to request a review 

---

##  **5. Stretch Tasks (Optional)**
- [ ] Tried a different **featuriser**  
- [ ] Tested other **ML architectures**  
- [ ] Applied the model to **public datasets (e.g., ChEMBL)**  
- [ ] Any other improvements? (Describe below)  

---

## 📌 **Requesting Review**
Once you’ve completed **Steps 1-4**, request a review by mentioning a mentor:  
👤 *@GemmaTuron* Please review my work!  

---

### 📝 **Asking questions**
Use this issue to update your progress, explain where you are stuck or ask for review!

Thank you for contributing! 🎉

**Comment by abdullahakintobi** [comment]
### Motivation letter

Dear Gemma,

I am excited to be selected by Outreachy for the participation period of the internship. My motivation for applying for this Outreachy internship program is to enhance my soft and technical Machine Learning skills for real-life application. I also selected the Ersilia Open Source Initiative for the participation phase because their mission to equip laboratories in low and middle-income countries with state-of-the-art AI/ML tools for infectious and neglected disease research align perfectly with my goal of solving global health issues with Machine Learning and Artificial Intelligence. 

As an experienced Data Analyst with a background in statistics and a fundamental knowledge of ML, I have developed strong analytical, problem-solving, and critical thinking skills and proficiency in programming languages such as SQL and Python, tools and libraries such as Pandas, SciPy, matplotlib, seaborn, scikit-learn, TensorFlow, and PyTorch, and statistical techniques such as correlation analysis, regression analysis, bayesian statistics, causal inference and many more. 

Ersilia's mission and goals align with my career goals—from facilitating the adoption of artificial intelligence and machine learning (AI/ML) models for biomedical research to the support of research into infectious and neglected diseases in LMICs. I have always been enthused by solving various world issues with AI/ML. Even though I major in statistics, I have been committed to reading research papers on medicine and the implementation of data science in the medical domain. My curiosity led me to read deeply about human biology and anatomy—and how data science concepts can be implemented in the domain. I am passionate about solving real-life medical issues with AI and ML and contributing to the biomedical science domain.

I am committed to continuous learning and enthused about being part of the Ersilia Open Source and making valuable contribution to the amazing project. Becoming part of the project will help me enhance my AI and ML skills, learn from industry experts and professionals, and collaborate with great and like minds in the industry. During the internship period, I will be able to have a more in-depth understanding of AI/ML, its processes, and its implementation in biomedical. The internship helps me improve my already established collaboration and team working skills. The completion of the internship will greatly impact my career by giving me an advantage in the understanding of the AI/ML process at the industry level from production to testing and implementation. As a data analyst with a strong passion for machine learning, artificial intelligence, and data science implementation in medicine, this internship will help my career by providing me with the experience needed to prosper in the job market. 

I am highly motivated to contribute to Ersilia’s mission and believe that my analytical mindset, problem-solving abilities, commitment to continuous learning, teamwork and collaboration skills will make me a valuable addition to the Ersilia team. I look forward to the possibility of joining this initiative and making meaningful contributions while growing and upskilling as a data scientist.

Thank you so much for your time and consideration. I appreciate the opportunity to apply and look forward to the possibility of contributing to this impactful project.

Sincerely,
Abdullah Akintobi

**Comment by abdullahakintobi** [comment]
# Installed the Ersilia Model Hub and successfully run models!

I tested the Ersilia Model Hub by fetching and serving the models `eos3b5e`, `eos2r5a`, and `eos9ei3`. The command executed successfully, and I obtained predictions without errors. This confirms that my installation works appropriately.

## For the `eos3b5e` model, I obtained as follows:
### Input:
```bash
# fetch the model from the server
ersilia fetch eos3b5e
# serve the model
ersilia serve eos3b5e
# run and test the model using the SMILES representation of Ibuprofen
ersilia run -i "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O"
```
### Output:
```json
[
    {
        "input": {
            "key": "HEFNNWSXXWATRW-UHFFFAOYSA-N",
            "input": "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O",
            "text": "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O"
        },
        "output": {
            "outcome": [
                206.28499999999997
            ]
        }
    }
]
```

## For the `eos2r5a` model, I obtained as follows:
### Input:
```bash
# fetch the model from the server
ersilia fetch eos2r5a
# serve the model
ersilia serve eos2r5a
# run and test the model using the SMILES representation of Ibuprofen
ersilia run -i "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O"
```
### Output:
```json
[
    {
        "input": {
            "key": "HEFNNWSXXWATRW-UHFFFAOYSA-N",
            "input": "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O",
            "text": "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O"
        },
        "output": {
            "outcome": [
                0.9899624
            ]
        }
    }
]
```

## For the `eos9ei3` model, I obtained as follows:
### Input:
```bash
# fetch the model from the server
ersilia fetch eos9ei3
# serve the model
ersilia serve eos9ei3
# run and test the model using the SMILES representation of Ibuprofen
ersilia run -i "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O"
```
### Output:
```json
[
    {
        "input": {
            "key": "HEFNNWSXXWATRW-UHFFFAOYSA-N",
            "input": "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O",
            "text": "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O"
        },
        "output": {
            "outcome": [
                2.191755
            ]
        }
    }
]
```

**Comment by abdullahakintobi** [comment]
### Recorded system settings

- Ubuntu 24.04.2 LTS
- Python 3.12.9
- Docker 27.5.1
- Conda 25.1.1

**Comment by abdullahakintobi** [comment]
### Explained completed tests of Ersilia

This issue details the process I follow to install the Ersilia Model Hub and successfully run and test the required models (`eos3b5e`, `eos2r5a`, and `eos9ei3`).

## 1. Installed the Ersilia Model Hub
After all the prerequisites according to the Ersilia Book were met, I proceeded to install the Ersilia tool by opening my WSL Ubuntu terminal and setting up a Conda environment.

### Created and Activated a Conda Environment
Ersilia requires a Conda environment for proper installation and execution. I set up the environment using:
```bash
conda create -n ersilia python=3.12
conda activate ersilia
```
**Purpose:** This ensures that Ersilia runs in an isolated Python environment without conflicts from other dependencies.

Then, I installed the Ersilia Python package.
```bash
git clone https://github.com/ersilia-os/ersilia.git
cd ersilia
pip install -e .
```
I then checked that the CLI works on my terminal, and explored the following commands
```bash
ersilia --help
ersilia catalog
```

## 2. Ran the Ersilia Models
After I successfully installed Ersilia, I proceeded to run the required models by opening WSL on VS Code via the Ubuntu terminal. I then changed my working environment with `conda` command to the ersilia environment and changed my working directory to the ersilia folder.

## 2.1. For the Model `eos3b5e`
### Step 1: Fetched the Model
To fetch the `eos3b5e` model, I ran:
```bash
ersilia fetch eos3b5e
```
**Purpose:** This command downloads the model and prepares it for execution.

### Step 2: Serving the Model
After fetching, I served the model by running:
```bash
ersilia serve eos3b5e
```
**Purpose:** Serving the models allows them to be used interactively via API calls.

### Step 3: Ran the Model with SMILES Input
The model was tested using the SMILES representation of Ibuprofen:
```bash
ersilia run -i "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O"
```
**Purpose:** This sends the Ibuprofen SMILES string as input to the model for processing.

### Step 4: Output
Upon execution, the model returned a JSON response containing the result:
```json
[
    {
        "input": {
            "key": "HEFNNWSXXWATRW-UHFFFAOYSA-N",
            "input": "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O",
            "text": "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O"
        },
        "output": {
            "outcome": [
                206.28499999999997
            ]
        }
    }
]
```

## 2.2. For the Model `eos2r5a`
### Step 1: Fetched the Model
To fetch the `eos2r5a` model, I ran:
```bash
ersilia fetch eos2r5a
```
**Purpose:** This command downloads the model and prepares it for execution.

### Step 2: Serving the Model
After fetching, I served the model by running:
```bash
ersilia serve eos2r5a
```
**Purpose:** Serving the models allows them to be used interactively via API calls.

### Step 3: Ran the Model with SMILES Input
The model was tested using the SMILES representation of Ibuprofen:
```bash
ersilia run -i "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O"
```
**Purpose:** This sends the Ibuprofen SMILES string as input to the model for processing.

### Step 4: Output
Upon execution, the model returned a JSON response containing the result:
```json
[
    {
        "input": {
            "key": "HEFNNWSXXWATRW-UHFFFAOYSA-N",
            "input": "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O",
            "text": "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O"
        },
        "output": {
            "outcome": [
                0.9899624
            ]
        }
    }
]
```

## 2.3. For the Model `eos9ei3`
### Step 1: Fetched the Model
To fetch the `eos9ei3` model, I ran:
```bash
ersilia fetch eos9ei3
```
**Purpose:** This command downloads the model and prepares it for execution.

### Step 2: Serving the Model
After fetching, I served the model by running:
```bash
ersilia serve eos9ei3
```
**Purpose:** Serving the models allows them to be used interactively via API calls.

### Step 3: Ran the Model with SMILES Input
The model was tested using the SMILES representation of Ibuprofen:
```bash
ersilia run -i "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O"
```
**Purpose:** This sends the Ibuprofen SMILES string as input to the model for processing.

### Step 4: Output
Upon execution, the model returned a JSON response containing the result:
```json
[
    {
        "input": {
            "key": "HEFNNWSXXWATRW-UHFFFAOYSA-N",
            "input": "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O",
            "text": "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O"
        },
        "output": {
            "outcome": [
                2.191755
            ]
        }
    }
]
```

## Conclusion
Following these steps, I successfully installed Ersilia. I fetched, served, and executed the required models with valid SMILES input. The results confirmed that the models were functioning correctly.

**Comment by abdullahakintobi** [comment]
👤 @GemmaTuron Please review my work!

**Comment by GemmaTuron** [comment]
Hello @abdullahakintobi 


Thanks for this good start to the contribution period, I loved hearing about your natural interest for medicine and AI
Please go ahead and register your first contribution for Ersilia in the Outreachy page linking this issue, where we will continue working during the rest of the internship. If you can, join in the welcome call today at 17:00 CET for more info!

**Comment by GemmaTuron** [comment]
Hi @abdullahakintobi 

Do you still plan on making an application to Ersilia? if so I suggest you start on w2 and 3 tasks as soon as possible. Else please close this issue

**Comment by abdullahakintobi** [comment]
Dear @GemmaTuron,

Thank you for the opportunity to participate in this project. Unfortunately, I was hospitalized and unable to meet the deadline. I appreciate the experience and the chance to connect with like-minded individuals and learn from industry experts like yourself. To focus on my recovery, I have closed this issue.

Thank you again for your consideration.

Best regards,

**Comment by GemmaTuron** [comment]
Hi @abdullahakintobi 

I am sorry to hear this, I hope you recover quickly and looking forward to having you in future rounds.
